protocol/x/vault/keeper/msg_server_deposit_to_vault.go

```diff
@@ -32,13 +34,16 @@ func (k msgServer) DepositToVault(
 	// Transfer from sender subaccount to vault.
 	// Note: Transfer should take place after minting shares for
 	// shares calculation to be correct.
+	if !quoteQuantums.IsUint64() {
```

Comment:
Does this cause the transaction to fail? Can you do this up-front instead? Is there some validateBasic function this can go in?